File: go/PivotInteger/PivotInteger.go
package main

import "fmt"

// Time -> 0(N) Space -> 0(1)
func pivotInteger(n int) int {
	totSum := 0
	start := 1

	for start <= n {
		totSum += start
		start++
	}

	start = 1
	runSum := 0
	for start <= n {
		runSum += start
		if runSum == totSum {
			return start
		}
		totSum -= start
		start++
	}

	return -1
}

func main() {
	fmt.Println(pivotInteger(4))
}

File: python/pivotInteger.py
# Time -> 0(N) Space -> 0(1)
def pivotInteger(n: int) -> int:
    totSum = 0
    for idx in range(1, n + 1):
        totSum += idx

    runSum = 0
    for idx in range(1, n + 1):
        runSum += idx
        if runSum == totSum:
            return idx
        totSum -= idx

    return -1


print(pivotInteger(4))
